src/App.js

```diff
@@ -1,13 +1,46 @@
-import React from 'react';
-
+import React, { useState, useEffect } from 'react';
 import './App.scss';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { GoodsList } from './Components/GoodsList/GoodsList';
+
+const App = () => {
+  const handleGoods = async(func) => {
+    const arr = await func();
+
+    setGoods(arr);
+  };
+
+  const [goods, setGoods] = useState([]);
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+  useEffect(() => {
+    handleGoods(getAll);
+  }, []);
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+  return (
+    <>
+      <div>
+        <button
+          type="submit"
+          onClick={() => handleGoods(getAll)}
+        >
+          all
+        </button>
+        <button
+          type="submit"
+          onClick={() => handleGoods(getRedGoods)}
+        >
+          red
+        </button>
+        <button
+          type="submit"
```

Comment:
```suggestion
          type="button"
```
Эти кнопки не внутри формы, так что думаю тут хватит просто button

Reply:
понял, сейчас исправлю